### PR TITLE
test: add test for logout

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Add test for logout endpoint

### CODE REVIEW

File: api/logout_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	require.NoError(ts.T(), ts.API.db.Create(u), "Error saving new test user")
+
+	// generate access token to use for logout
+	t, err := generateAccessToken(u, nil, time.Second*time.Duration(ts.Config.JWT.Exp), ts.Config.JWT.Secret)
+	require.NoError(ts.T(), err)
+	ts.token = t
+}
+
+func (ts *LogoutTestSuite) TestLogoutSuccess() {
+	req := httptest.NewRequest(http.MethodPost, "http://localhost/logout", nil)
```
Comment (hf) [inline api/logout_test.go:50]: Typically logout is a DELETE request. (Just making a not of it.)

File: api/logout_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ts.token))
+	w := httptest.NewRecorder()
+
+	ts.API.handler.ServeHTTP(w, req)
+	require.Equal(ts.T(), http.StatusNoContent, w.Code)
```
Comment (hf) [inline api/logout_test.go:55]: Can you check that the cookies are deleted in the response? I think that's the important bit.

File: api/logout_test.go (hunk 1 above)
Comment (kangmingtay) [reply api/logout_test.go:50]: hmm really? both auth0 and aws cognito's logout endpoint are using the `GET` method but i can see why that's undesirable in the case of accidental pre-fetching which may cause a user to logout before the `/logout` endpoint is actually called. 

also worth noting that as defined by mdn, the `DELETE` method should be [idempotent](https://developer.mozilla.org/en-US/docs/Glossary/Idempotent) while `POST` methods aren't

**Review by hf — APPROVED** [review]
LGTM!

**Pull request merged**